notifikasjon/src/main/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/notifikasjon/river/OpprettForespoerselSakOgOppgaveRiver.kt

```diff
@@ -0,0 +1,115 @@
+import no.nav.helsearbeidsgiver.felles.EventName
+import no.nav.helsearbeidsgiver.felles.Key
+import no.nav.helsearbeidsgiver.felles.domene.Person
+import no.nav.helsearbeidsgiver.felles.json.krev
+import no.nav.helsearbeidsgiver.felles.json.les
+import no.nav.helsearbeidsgiver.felles.json.toJson
+import no.nav.helsearbeidsgiver.felles.json.toMap
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.model.Fail
+import no.nav.helsearbeidsgiver.felles.rapidsrivers.river.ObjectRiver
+import no.nav.helsearbeidsgiver.felles.utils.Log
+import no.nav.helsearbeidsgiver.inntektsmelding.notifikasjon.NotifikasjonTekst
+import no.nav.helsearbeidsgiver.inntektsmelding.notifikasjon.opprettOppgave
+import no.nav.helsearbeidsgiver.inntektsmelding.notifikasjon.opprettSak
+import no.nav.helsearbeidsgiver.utils.json.serializer.UuidSerializer
+import no.nav.helsearbeidsgiver.utils.json.toJson
+import no.nav.helsearbeidsgiver.utils.log.logger
+import no.nav.helsearbeidsgiver.utils.log.sikkerLogger
+import no.nav.helsearbeidsgiver.utils.wrapper.Orgnr
+import java.util.UUID
+
+data class OpprettForespoerselSakOgOppgaveMelding(
+    val eventName: EventName,
+    val transaksjonId: UUID,
+    val forespoerselId: UUID,
+    val orgnr: Orgnr,
+    val sykmeldt: Person,
+    val orgNavn: String,
+)
+
+class OpprettForespoerselSakOgOppgaveRiver(
+    private val lenkeBaseUrl: String,
+    private val agNotifikasjonKlient: ArbeidsgiverNotifikasjonKlient,
+) : ObjectRiver<OpprettForespoerselSakOgOppgaveMelding>() {
+    private val logger = logger()
+    private val sikkerLogger = sikkerLogger()
+
+    override fun les(json: Map<Key, JsonElement>): OpprettForespoerselSakOgOppgaveMelding? =
+        if (setOf(Key.BEHOV, Key.FAIL).any(json::containsKey)) {
+            null
+        } else {
+            val data = json[Key.DATA]?.toMap().orEmpty()
+
+            OpprettForespoerselSakOgOppgaveMelding(
+                eventName = Key.EVENT_NAME.krev(EventName.SAK_OG_OPPGAVE_OPPRETT_REQUESTED, EventName.serializer(), json),
+                transaksjonId = Key.UUID.les(UuidSerializer, json),
+                forespoerselId = Key.FORESPOERSEL_ID.les(UuidSerializer, data),
+                orgnr = Key.ORGNRUNDERENHET.les(Orgnr.serializer(), data),
+                sykmeldt = Key.SYKMELDT.les(Person.serializer(), data),
+                orgNavn = Key.VIRKSOMHET.les(String.serializer(), data),
+            )
+        }
+
+    override fun OpprettForespoerselSakOgOppgaveMelding.haandter(json: Map<Key, JsonElement>): Map<Key, JsonElement> {
+        val lenke = NotifikasjonTekst.lenkeAktiv(lenkeBaseUrl, forespoerselId)
+
+        val sakId =
+            agNotifikasjonKlient.opprettSak(
+                lenke = lenke,
+                inntektsmeldingTypeId = forespoerselId,
+                orgnr = orgnr,
+                sykmeldt = sykmeldt,
+            )
+
+        val oppgaveId =
+            agNotifikasjonKlient.opprettOppgave(
```

Comment:
Kan vi havne i en situasjon der saken ble opprettet men ikke oppgaven? Vil vi da forsøke på nytt og lage en helt ny sak og en helt ny oppgave?

Reply:
Du er inne på noe, men pga. at feillytteren ikke tåler meldinger uten behov så vil ikke en feilet oppgaveopprettelse føre til at man prøver på ny. Da står vi uten oppgave, men siden det er sjeldent at kallene mot fager feiler så kan vi leve med risikoen i en liten stund, til vi fikser feillytteren. Før vi gjør det så må vi imidlertid tåle duplikater, som er hvorfor jeg roter inne i notifikasjonsklienten.

Vi har uansett fjernet den største synderen for at saker og oppgaver ikke blir opprettet, som var de gamle servicene.